### ISSUE TITLE
PR build not using merged shippable.yml to generate matrix

### ISSUE THREAD

**Issue by mattclay** [Shippable/support#2643]
It appears PR builds use a build matrix without first merging with HEAD of the base branch.

For example: https://app.shippable.com/runs/5743411ba7f3850d00dfd5a7

This PR was created using a branch which doesn't have shippable.yml (only a .travis.yml). However, the branch the PR is to be merged with does have a shippable.yml. My expectation was that the PR branch would be merged with HEAD of the base branch and the shippable.yml would be used from that.

Instead, no shippable.yml was used and instead the build matrix was generated from the .travis.yml.

**Comment by manishas** [comment]
@rageshkrishna can you look into this?

**Comment by rageshkrishna** [comment]
Hi @mattclay. It looks like you've run in to a scenario that our YML merge design doesn't accommodate. We currently assume that both branches will contain the same variant of the YML file (either travis.yml or shippable.yml).

We'll definitely look in to fixing this, but in the short term the only resolution is to have a shippable.yml file in the branch.

**Comment by abhijitkini** [comment]
@mattclay closing this issue as resolved, since there is a short term resolution of having the shippable.yml file in the branch. We are looking into the feature request of adding it as a project setting in the UI.

**Comment by mattclay** [comment]
Having the project setting would be very useful. We're having problems with this again now that we just enabled another two repositories which are also using Travis.

The "short-term resolution" above only works if we can communicate the requirement to all of our contributors before submitting a PR, which isn't practical. What ends up happening is we wait until Shippable fails tests for a PR and then provide instructions to the contributor on how to rebase their branch to resolve the issue.

The problem will lessen over time as fewer PRs are submitted based on older forks of our code, which were made before we enabled Shippable. However, until that happens this makes the transition to Shippable more disruptive than it needs to be.
